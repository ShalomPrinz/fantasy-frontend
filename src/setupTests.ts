// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import "@testing-library/jest-dom";
import axios from "axios";
import { setupServer } from "msw/node";

import { handlers } from "../test";
import * as TeamContext from "./contexts/TeamContext";
import * as UserContext from "./features/authentication/contexts/UserContext";
import { LeagueInfo, Message, Player, Team, User } from "./types";

jest.setTimeout(10000);

axios.defaults.baseURL = "";

const server = setupServer(...handlers);
beforeAll(() => server.listen());
afterEach(() => server.resetHandlers());
afterAll(() => server.close());

interface MockedUserContext {
  logout?: () => Promise<void>;
  state?: UserContext.UserState;
  user?: User;
}

interface MockedUser {
  inbox?: Message[];
  leagues?: LeagueInfo[];
  name?: string;
  players?: Player[];
}

export const getTestUser = (user?: MockedUser): User => ({
  id: "id",
  inbox: user?.inbox || [],
  leagues: user?.leagues || [],
  name: user?.name || "",
  team: new Team(user?.players || []),
});

export const mockUser = (context: MockedUserContext) => {
  const userContext = jest.spyOn(UserContext, "useUser");

  const { logout, state, user } = context;
  const mockedLogout = typeof logout !== "undefined" ? logout : async () => {};
  const mockedState =
    typeof state !== "undefined" ? state : UserContext.UserState.LOGGED_USER;
  const mockedUser = typeof user !== "undefined" ? user : getTestUser();

  userContext.mockImplementation(() => ({
    loading: false,
    login: async () => {},
    logout: mockedLogout,
    register: async () => {},
    state: mockedState,
    user: mockedUser,
    invalidate: async () => {},
  }));

  return userContext.mockRestore;
};

export const mockTeam = (team: Team) => {
  const teamContext = jest.spyOn(TeamContext, "useTeamState");
  teamContext.mockImplementation(() => team);
};

export * from "../test";
