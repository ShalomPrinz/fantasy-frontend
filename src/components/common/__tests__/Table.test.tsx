import { render } from "setupTests";

import { Table } from "../../";

describe("Table", () => {
  it("should render Table component", () => {
    const data = [{ id: 1, label: "Label" }];
    const columns = [{ id: 0, path: "label" }];

    const { asFragment } = render(<Table data={data} columns={columns} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("should render a column by content callback", () => {
    const data = [{ id: 1, label: "Label" }];
    const content = ({ label }: { [key: string]: string }) => <h1>{label}</h1>;
    const columns = [{ id: 0, path: "label", content: content }];

    const { asFragment } = render(<Table data={data} columns={columns} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
