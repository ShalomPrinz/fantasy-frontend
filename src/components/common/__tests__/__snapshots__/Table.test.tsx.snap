// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table should render a column by content callback 1`] = `
<DocumentFragment>
  <div
    class="table-responsive mt-0"
  >
    <table
      class="table rounded bg-default"
    >
      <tbody>
        <tr>
          <td
            class="align-middle"
          >
            <h1>
              Label
            </h1>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;

exports[`Table should render a scrollable table 1`] = `
<DocumentFragment>
  <div
    class="table-responsive mt-0 border-bottom scrollable"
  >
    <table
      class="table rounded"
    >
      <tbody>
        <tr>
          <td
            class="align-middle"
          >
            Label
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;

exports[`Table should render a styled table 1`] = `
<DocumentFragment>
  <div
    class="table-responsive mt-0"
  >
    <table
      class="table rounded bg-default"
    >
      <tbody>
        <tr>
          <td
            class="align-middle"
          >
            Label
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;
